carts-core/src/test/java/io/helidon/examples/sockshop/carts/TestCartRepository.java

```diff
@@ -0,0 +1,29 @@
+package io.helidon.examples.sockshop.carts;
+
+/**
+ * Tests may need additional methods in repository implementations
+ * in order to pre-load test data, clear the repository between the
+ * test, etc.
+ *
+ * <p>The methods needed only for the tests should reside in the test
+ * project, to avoid polluting production repository implementation
+ * with methods it a) doesn't need, and b) may be downright dangerous.
+ *
+ * <p>Note that for the CDI to work it is necessary to mark the test
+ * repositories with @Alternative and @Priority exceeding that of the
+ * implementation.
+ *
+ * <p>Note that for the tests to test the actual implementation it is
+ * necessary that the test implementations only extend the repository
+ * class and add methods, but do not override any methods from the
+ * repository implementation they are testing.
+ *
+ * @author Aleks Seovic  2020.05.13
+ */
+public interface TestCartRepository extends CartRepository {
```

Comment:
What do you need this test repository for? We have a way to remove cart by id, which is what all the tests use, so we don't need test repository in this service -- nothing is using it, and it doesn't exist in `carts-core` or any other backend implementation.